### PR TITLE
Fix the bug that cascaded tables were dropped before `save()` is called

### PR DESCRIPTION
**Pull request by d-hayashi**
## What?
Fix the bug that `DatabaseIDDBHandler.remove_data` drops the table which stores the metadata of the corresponding database-id before executing `save()`.

## Why?
Bug fix

## Related issues
#81 

## Note
Test case `_test_remove_database_id` is added for this fix, but disabled as PyDTK does not support dropping tables with TinyDB < 4.  
This should be enabled when #83 is resolved.

### CODE REVIEW

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/pydtk/pull/84#" title="Author self-approved">d-hayashi</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fpydtk).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/pydtk/blob/master/OWNERS)~~ [d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by sudach — APPROVED** [review]
/lgtm

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/pydtk/pull/84#" title="Author self-approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/pydtk/pull/84#pullrequestreview-720754485" title="LGTM">sudach</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fpydtk).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/pydtk/blob/master/OWNERS)~~ [d-hayashi,sudach]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
